synthesizer/src/store/helpers/mod.rs

```
@@ -111,17 +111,14 @@ pub trait MapRead<
         K: Borrow<Q>,
         Q: PartialEq + Eq + Hash + Serialize + ?Sized,
     {
-        // Return early in case of errors in order to not conceal them.
```

The comment states to return early.

---

Yes, which was significant in the older logic, where the map value was the one to use if there was one (instead of ignoring the error in case there was a batched one available); however, now that the batch value is the one to be returned if there is one, there is no need to look up the map value in all the calls.

---

Understood, that makes sense. Let me review Ray's PR on finalize (which uses this code heavily) first, and assuming all is good, I'll come back and get this landed.